src/groupByKey.ts

```diff
@@ -2,6 +2,19 @@ type GroupsMap<T> = {
   [key: string]: T[];
 };
 
-export function groupByKey(items, key) {
-  // write code here;
+export function groupByKey(items: any[], key: string): any {
+  const groupedObj: GroupsMap<any> = {};
+
+  items.map((item) => {
+    if (groupedObj[item[key]] === undefined) {
```

Comment:
It is better to check if property is present in object using "[in](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Operators/in)" or "[hasOwnProperty](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Object/hasOwnProperty)"